commercetools/resource_category.go

```diff
@@ -0,0 +1,284 @@
+		newName := commercetools.LocalizedString(expandStringMap(d.Get("name").(map[string]interface{})))
+		input.Actions = append(
+			input.Actions,
+			&commercetools.CategoryChangeNameAction{Name: &newName})
+	}
+
+	if d.HasChange("slug") {
+		newSlug := commercetools.LocalizedString(expandStringMap(d.Get("slug").(map[string]interface{})))
+		input.Actions = append(
+			input.Actions,
+			&commercetools.CategoryChangeSlugAction{Slug: &newSlug})
+	}
+
+	if d.HasChange("key") {
+		newKey := d.Get("key").(string)
+		input.Actions = append(
+			input.Actions,
+			&commercetools.CategorySetKeyAction{Key: newKey})
+	}
+
+	if d.HasChange("order_hint") {
+		newVal := d.Get("order_hint").(string)
+		input.Actions = append(
+			input.Actions,
+			&commercetools.CategorySetKeyAction{Key: newVal})
+	}
+
+	if d.HasChange("external_id") {
+		newVal := d.Get("external_id").(string)
+		input.Actions = append(
+			input.Actions,
+			&commercetools.CategorySetKeyAction{Key: newVal})
+	}
+
+	if d.HasChange("description") {
+		newDescription := commercetools.LocalizedString(expandStringMap(d.Get("description").(map[string]interface{})))
+		input.Actions = append(
+			input.Actions,
+			&commercetools.CategorySetDescriptionAction{Description: &newDescription})
+	}
+
+	if d.HasChange("parent") {
+		newParentCategoryId := d.Get("parent").(string)
+		parentId := commercetools.CategoryResourceIdentifier{ID: newParentCategoryId}
+		input.Actions = append(
+			input.Actions,
+			&commercetools.CategoryChangeParentAction{Parent: &parentId})
+	}
+
+	if d.HasChange("meta_title") {
+		newMetaTitle := commercetools.LocalizedString(expandStringMap(d.Get("meta_title").(map[string]interface{})))
+		input.Actions = append(
+			input.Actions,
+			&commercetools.CategorySetMetaTitleAction{MetaTitle: &newMetaTitle})
+	}
+
+	if d.HasChange("meta_description") {
+		newMetaDescription := commercetools.LocalizedString(expandStringMap(d.Get("meta_description").(map[string]interface{})))
+		input.Actions = append(
+			input.Actions,
+			&commercetools.CategorySetMetaDescriptionAction{MetaDescription: &newMetaDescription})
+	}
+
+	if d.HasChange("meta_keywords") {
+		newMetaKeywords := commercetools.LocalizedString(expandStringMap(d.Get("meta_keywords").(map[string]interface{})))
+		input.Actions = append(
+			input.Actions,
+			&commercetools.CategorySetMetaKeywordsAction{MetaKeywords: &newMetaKeywords})
+	}
+
+	log.Printf(
+		"[DEBUG] Will perform update operation with the following actions:\n%s",
+		stringFormatActions(input.Actions))
+
+	_, err = client.CategoryUpdateWithID(context.Background(), input)
+	if err != nil {
+		if ctErr, ok := err.(commercetools.ErrorResponse); ok {
+			log.Printf("[DEBUG] %v: %v", ctErr, stringFormatErrorExtras(ctErr))
+		}
+		return err
+	}
+
+	return resourceCategoryRead(d, m)
+}
+
+func resourceCategoryDelete(d *schema.ResourceData, m interface{}) error {
+	client := getClient(m)
+
+	category, err := client.CategoryGetWithID(context.Background(), d.Id())
```

Comment:
There's no need for this call. You can get the version from the state like so:
```suggestion
	version := d.Get("version").(int)
```
And use that in the `client.CategoryDeleteWithID`. Should be a bit faster